cluster_domain/external_dns.tf

```diff
@@ -84,3 +87,19 @@ resource "helm_release" "external_dns_identity" {
     value = var.plugin.cluster_identity_client_id
   }
 }
+
+data "azurerm_kubernetes_cluster" "cluster" {
+  name                = var.plugin.cluster_name
+  resource_group_name = var.plugin.cluster_resource_group_name
+}
+
+
+resource "azurerm_federated_identity_credential" "identity_credential" {
+  parent_id           = var.plugin.cluster_identity_id
+  name                = "external-dns-access"
+  resource_group_name = var.plugin.cluster_resource_group_name
+
+  audience = ["api://AzureADTokenExchange"]
+  subject  = "system:serviceaccount:${kubernetes_namespace.external_dns.metadata[0].name}:external-dns"
+  issuer   = data.azurerm_kubernetes_cluster.cluster.oidc_issuer_url
```

Comment:
I wonder if we shouldn't pass oidc_issuer_url directly instead of relying on datasource?

Reply:
Well, I think in the larger scale it's easier for submodules to fetch what they need. What are the downsides?